src/Avalonia.Controls/Platform/ITopLevelImpl.cs

```diff
@@ -141,6 +141,11 @@ public interface ITopLevelImpl : IOptionalFeatureProvider, IDisposable
         /// </summary>
         Action? Closed { get; set; }
         
+        /// <summary>
+        /// Gets or sets a method called right before the underlying implementation is destroyed.
+        /// </summary>
```

Comment:
The modern .NET naming convention would be `Closing` I think. That's what is used elsewhere on events. I realize this is not an event but it probably makes sense to use the standard naming convention.

Reply:
We have separate `Closing` (cancelable) event in `IWindowImpl`

Reply:
Hm, well, it's up to you if you think it is too confusing. This might also get into app lifecycle events and there is no standard for that between MAUI/UWP/WinUI/WPF.